### ISSUE TITLE
Error when training with semantic segmentation 

### ISSUE THREAD

**Issue by nr1040gh** [azavea/raster-vision#1010]
Hi, I am new to RasterVision and machine learning/ai in general and I am trying to follow one of the examples for semantic segmentation with raster labels (https://github.com/azavea/raster-vision/blob/0.12/rastervision_pytorch_backend/rastervision/pytorch_backend/examples/semantic_segmentation/isprs_potsdam.py). However, when I implement this, it crashes right at the train command and I receive the exact same error as posted in (https://github.com/azavea/raster-vision/issues/969). I have this error for data I provided as well as the potsdam data used in the example. I have included my code down below. I've been trying to remedy this with no success - if my LabelSource returns labels with class_ids that are not in [0, num_classes), is there anyway to see what they are returning and or what the interval [0, num_classes) is?

Thank you

```python
import os

from os.path import join

from rastervision.core.rv_pipeline import *
from rastervision.core.backend import *
from rastervision.core.data import *
from rastervision.pytorch_backend import *
from rastervision.pytorch_learner import *
from rastervision.core.analyzer import *

def get_config(runner):
	root_uri = '/opt/data/output'
	base_uri = '/opt/data/potsdam/test'
	gtlabel_image1 = base_uri + '/top_potsdam_2_10_label.tif'
	image1 = base_uri + '/top_potsdam_2_10_RGBIR.tif'

	gtlabel_image2 = base_uri + '/top_potsdam_7_12_label.tif'
	image2 = base_uri + '/top_potsdam_7_12_RGBIR.tif'

	class_config = ClassConfig(names=['Car', 'Building', 'Low Vegetation', 'Tree', 'Impervious','Clutter'],colors=['#ffff00', '#0000ff', '#00ffff', '#00ff00', '#ffffff', '#ff0000'])

	channel_order = [3,0,1]
	def make_scene(scene_id,image_uri,label_uri,root_uri):
        # infrared, red, green
		raster_source = RasterioSourceConfig(uris=[image_uri],channel_order=channel_order)
		# Using with_rgb_class_map because label TIFFs have classes encoded as
		# RGB colors.
		label_source = SemanticSegmentationLabelSourceConfig(rgb_class_config=class_config,raster_source=RasterioSourceConfig(uris=[label_uri]))

		# URI will be injected by scene config.
		# Using rgb=True because we want prediction TIFFs to be in
		# RGB format.
		label_store = SemanticSegmentationLabelStoreConfig(rgb=True, vector_output=[PolygonVectorOutputConfig(class_id=0)])

		scene = SceneConfig(
		    id=scene_id,
		    raster_source=raster_source,
		    label_source=label_source,
		    label_store=label_store)

		return scene

	dataset = DatasetConfig(
	    class_config=class_config,
	    train_scenes=[make_scene('2_10',image1,gtlabel_image1,root_uri)],
	    validation_scenes=[make_scene('7_12',image2,gtlabel_image2,root_uri)])
	chip_sz = 300
	chip_options = SemanticSegmentationChipOptions(window_method=SemanticSegmentationWindowMethod.random_sample,
		chips_per_scene=10)

	backend = PyTorchSemanticSegmentationConfig(
	    model=SemanticSegmentationModelConfig(backbone=Backbone.resnet50),
	    solver=SolverConfig(
	        lr=1e-4,
	        num_epochs=1,
	        batch_sz=3,
	        one_cycle=True))
	    #log_tensorboard=True,
	    #run_tensorboard=False,
	    #test_mode=test)

	return SemanticSegmentationConfig(
	    root_uri=root_uri,
	    dataset=dataset,
	    backend=backend,
	    train_chip_sz=chip_sz,
	    predict_chip_sz=chip_sz,
	    chip_options=chip_options)
```

**Comment by lewfish** [comment]
Did you run the Potsdam example as described in the example docs, and did you get an error then? It looks like the code you pasted above is based on the Potsdam example and uses the same data. I'm trying to figure out what is the change you made that causes the problem (if it's the case that the Potsdam example works).

**Comment by nr1040gh** [comment]
Hi, @lewfish, I eventually got past this error yesterday and received results. I ended up deleting everything rastervision put on my machine locally (__pycache__, pipeline.json, etc) and running the code again and I no longer run into this error and get past the training stage (for 1 epoch). I wonder if this has to do with files not being updated/replaced when I changed my code around? I am running the code using the instructions on the QuickStart guide, by first mounting all of my data to the docker directories since I have everything locally.

Now I get an error when I try to extend to multiple epochs (1 ->3):

INFO:rastervision.pytorch_learner.learner:epoch: 1
Training  [##################------------------]   50%/opt/data/output/Makefile:12: recipe for target '2' failed
make: *** [2] Killed

Should I close this issue since I resolved my initial problem?

**Comment by lewfish** [comment]
The fact that the process was killed with no other error messages makes me think it's an out of memory error. Assuming you are running this locally, I would increase the amount of memory that Docker is using (through the settings UI) to 8GB if possible.

**Comment by nr1040gh** [comment]
@lewfish, I'm now starting my docker container with 
`docker run --rm -it --ipc=host --memory=5g --memory-swap=-1\
     -v ${CODE_DIR}:/opt/src/code  \
     -v ${OUT_DIR}:/opt/data/output \
     -v ${DATA_DIR}:/opt/data/AerialImageDataset \
     -v ${POTS_DIR}:/opt/data/potsdam \
     quay.io/azavea/raster-vision:pytorch-0.12 /bin/bash`
and I see that during training my ram usage goes to 100% (~7.6gb/7.6gb) or close when it does crash. This leads me to believe that you are right and I am running out of ram.

**Comment by AdeelH** [comment]
I believe this was the same OOM error that was fixed by #1115. This error seemed to only occur when training on CPU instead of GPU. Closing this now, since it has been fixed.
